.maintain/gitlab/check_line_width.sh

```diff
@@ -5,19 +5,20 @@
 set -e
 set -o pipefail
 
-BASE_ORIGIN="origin"
-BASE_BRANCH_NAME="master"
 LINE_WIDTH="120"
 GOOD_LINE_WIDTH="100"
-BASE_BRANCH="${BASE_ORIGIN}/${BASE_BRANCH_NAME}"
 
-git fetch ${BASE_ORIGIN} ${BASE_BRANCH_NAME} --depth 1
-git diff --name-only ${BASE_BRANCH} -- \*.rs | ( while read file
+if [ -z $CI_COMMIT_BEFORE_SHA ]; then
```

Comment:
since this is not a merge request on gitlab it won't be [set](https://docs.gitlab.com/ee/ci/variables/predefined_variables.html).

Reply:
I thought a merge request on github counts as that because it is all mirrored. So you say this is expected?

Reply:
it is expected, yes. pull requests are only branches on gitlab (not merge requests due to the nature of the mirroring).


i'm also not finding it easy to understand the meaning of:

> The previous latest commit present on a branch before a merge request.

which would be the "a branch"?
basically the `...` do the right thing and that hash would probably in the best case to the same.

Reply:
> i'm also not finding it easy to understand the meaning of:

A PR branched off a branch at some point of time (lets call this branch `master`). Then both add commits. Then after some time you start at the latest commit of `master` and `PR` and go backwards following all their parents until you hit a common hash. This is what `$CI_COMMIT_BEFORE_SHA` is. Basically the hash on `master` where you started your PR from.

> basically the ... do the right thing and that variable would probably in the best case do the same.

1. You can just omit the right side of the  `...`. This is equivalent to `left...HEAD`. This is what I am doing here.

2. I cannot to "just" the same thing because I do not know the left side. This is what this PR is all about. `$CI_COMMIT_BEFORE_SHA` would give me this left side. However, this is not working.  I need to resort to other measures to determine it then.